src/player.cpp

```diff
@@ -9161,6 +9161,17 @@ int player::amount_of(const itype_id &it) const
     return quantity;
 }
 
+int player::amount_worn(const itype_id &id) const
+{
+    int amount = 0;
+    for(auto &elem : g->u.worn) {
```

Comment:
Should refer to `this->worn`, not `g->u.worn` (but without the redundant `this`, only ` : worn ` is fine). If the function is called on an NPC object, it should examine the worn items of the NPC, not the items of the player character.